src/network/networkd-link.c

```diff
@@ -1692,6 +1726,15 @@ static void link_detach_from_manager(Link *link) {
         link_unref(set_remove(link->manager->links_requesting_uuid, link));
         link_clean(link);
 
+        if (link->alternative_names) {
+                char **n;
+
+                STRV_FOREACH(n, link->alternative_names)
+                        hashmap_remove(link->manager->links_by_name, *n);
```

Comment:
as a safety precaution I think we should use hashmap_remove_value() here. The only difference is that it also takes both key *and* value when removing, and only removes the entry when the key actually points to the indicated value. i.e use 

```c
hashmap_remove_value(link->manager->links_by_name, *n, link);
```

This might be useful in case the hash table ever gets out of sync, because of some bug.